### DISCUSSION TITLE
How can I exclude GIT in home directory?

### DISCUSSION THREAD

**Discussion by GasimGasimzada** [JanDeDobbeleer/oh-my-posh#1089] [Troubleshoot] [answered]
I have a `.git` directory in home directory with some commands in them. Since GIT also works for all subdirectories, any directory that I have in home directory will show the alchemy sign:

<img width="215" alt="Screen Shot 2021-10-20 at 12 18 00 PM" src="https://user-images.githubusercontent.com/807231/138055214-2292b045-3cba-4248-a057-c933f1dc3bc1.png">

Is there a way that I can exclude this? I tried `exclude_folders: ["~"]` but it didn't work.

**Comment by JanDeDobbeleer** [comment, marked as answer]
Interesting, as you could exclude the home folder (full path, not tilde) which works when you're actually inside `$HOME`, but not when you're deeper. My main question is why would you have a `.git` folder in your home folder? We could work around this by creating an `exclude_repos` property as that's the same when nested.

**Reply by GasimGasimzada** [reply]
By default, `~/.git` is in default path; so, I store some custom GIT commands there, which allows me to run custom GIT commands from anywhere. For example, I have a shell file `~/.git/git-clean-repo`, which allows me to run `git clean-repo` from all my repos.

**Reply by JanDeDobbeleer** [reply]
@GasimGasimzada is that a folder or repo?

**Reply by GasimGasimzada** [reply]
Just a folder.

**Reply by GasimGasimzada** [reply]
Do you think it would make sense to change the logic for GIT block to be based on actual GIT command check instead of checking the `.git` directory? Example pseudocode:

```
# If exit code is 0, it means there is a repo
if (git -C $CURRENT_PATH rev-parse) == 0 {
  show git block
}
```

**Reply by JanDeDobbeleer** [reply]
@GasimGasimzada the easiest solution is to use exclude_folders [here](https://github.com/JanDeDobbeleer/oh-my-posh/blob/37659aab709b0c7dff36603ba7ab8f6e642321a7/src/segment_git.go#L134) to validate if that root path should be skipped.

**Reply by GasimGasimzada** [reply]
for the time being, I moved my my GIT commands to another directory but this change will allow me to revert back to my old workflow of storing GIT commands in `~/.git` directory.

**Reply by JanDeDobbeleer** [reply]
https://github.com/JanDeDobbeleer/oh-my-posh/pull/1097

**Reply by GasimGasimzada** [reply]
Thank you! This discussion can be answered/closed.
